### PR TITLE
BUGFIX: SD-527 fix promises in combination with forEach in unit test for combo

### PR DESCRIPTION
**Pull request by lvkhanh**
Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [x] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [x] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [x] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [x] Migration guide (for major changes) is mentioned in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [x] Successful `extended test - examples`
- [x] Successful `extended test - storybook`
- [x] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)

# Related PRs

- gdc-app-components:
- gdc-analytical-designer:
- gdc-dashboards:

# Related Jira tasks

### CODE REVIEW

**Comment by lvkhanh** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1048.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

File: src/internal/components/pluggableVisualizations/comboChart/tests/PluggableComboChart.spec.tsx [hunk 1]
```diff
@@ -331,47 +328,31 @@ describe("PluggableComboChart", () => {
...
-            ],
-            [
-                AXIS.DUAL,
-                COLUMN_AREA_LINE,
-                multipleMetricsAndCategoriesReferencePoint,
-                [...COMBO_CHART_SUPPORTED_PROPERTIES[AXIS.DUAL], ...OPTIONAL_STACKING_PROPERTIES],
-            ],
+            [AXIS.PRIMARY, COLUMN_AREA_LINE, oneMetricAndCategoryAndStackReferencePoint],
+            [AXIS.SECONDARY, COLUMN_AREA_LINE, measuresOnSecondaryAxisAndAttributeReferencePoint],
+            [AXIS.DUAL, COLUMN_AREA_LINE, multipleMetricsAndCategoriesReferencePoint],
```
Comment (ptquang86) [inline src/internal/components/pluggableVisualizations/comboChart/tests/PluggableComboChart.spec.tsx]: COLUMN_AREA_LINE: this parameter is the same for 3 tests

File: src/internal/components/pluggableVisualizations/comboChart/tests/PluggableComboChart.spec.tsx [hunk 2]
```diff
@@ -331,47 +328,31 @@ describe("PluggableComboChart", () => {
...
-
-                    expect(axisType).toEqual(axis);
-                    expect(comboChart.getSupportedPropertiesList()).toEqual(expectedStackProperties);
-                });
+            (axis: string, supportedChartTypes: string[], refPoint: IReferencePoint) => {
+                return Promise.all(
+                    supportedChartTypes.map(async (chartType: VisualizationObject.VisualizationType) => {
+                        const mockProps = {
+                            ...defaultProps,
+                            pushData: jest.fn(),
```
Comment (ptquang86) [inline src/internal/components/pluggableVisualizations/comboChart/tests/PluggableComboChart.spec.tsx]: could we get rid of this mock function?

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1588/ : SUCCESS in 15m 30s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5211/ : SUCCESS in 12s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4095/ : SUCCESS in 20m 03s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5210/ : SUCCESS in 11s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4097/ : SUCCESS in 13m 54s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5214/ : SUCCESS in 12s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4101/ : SUCCESS in 5m 05s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5216/ : SUCCESS in 10s

**Comment by lvkhanh** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2229/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1590/ : SUCCESS in 11m 12s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5218/ : SUCCESS in 11s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4103/ : SUCCESS in 20m 58s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5217/ : SUCCESS in 13s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4117/ : SUCCESS in 9m 54s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5237/ : SUCCESS in 8s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4118/ : SUCCESS in 5m 40s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5238/ : SUCCESS in 14s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4119/ : FAILURE in 1m 18s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5239/ : SUCCESS in 8s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4120/ : FAILURE in 1m 36s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5240/ : SUCCESS in 17s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4121/ : FAILURE in 1m 22s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5241/ : SUCCESS in 8s

File: src/internal/components/pluggableVisualizations/comboChart/tests/PluggableComboChart.spec.tsx [hunk 3]
```diff
@@ -331,46 +328,28 @@ describe("PluggableComboChart", () => {
...
+            (axis: string, refPoint: IReferencePoint) => {
+                const comboChart = createComponent(defaultProps);
+                const refPoints: Promise<IExtendedReferencePoint>[] = COLUMN_AREA_LINE.map(
+                    (chartType: VisualizationObject.VisualizationType) => {
+                        refPoint.buckets.forEach((bucket: IBucket) => (bucket.chartType = chartType));
+                        return comboChart.getExtendedReferencePoint(refPoint);
+                    },
+                );
+
+                Promise.all(refPoints).then(function(extRefPoints: IExtendedReferencePoint[]) {
```
Comment (naibinh) [inline src/internal/components/pluggableVisualizations/comboChart/tests/PluggableComboChart.spec.tsx]: Follow this point from [here](https://jestjs.io/docs/en/asynchronous#promises)
`Be sure to return the promise - if you omit this return statement, your test will complete before the promise returned from fetchData resolves and then() has a chance to execute the callback.`
I think we should `return Promise.all(...)`

And please consider using `for of`
For example
```
for (const extRefPoint of extRefPoints) {
...
}
```

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4122/ : SUCCESS in 7m 01s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5242/ : SUCCESS in 8s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4124/ : SUCCESS in 4m 27s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5244/ : SUCCESS in 8s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4126/ : SUCCESS in 8m 16s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5246/ : SUCCESS in 7s

**Comment by naibinh** [comment]
LGTM

**Comment by lvkhanh** [comment]
I squashed commit

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4129/ : SUCCESS in 5m 07s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5248/ : SUCCESS in 9s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2239/ : SUCCESS in 17m 46s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4130/ : SUCCESS in 13m 10s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1071/ : SUCCESS in 30s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/1271/ : SUCCESS in 31m 58s

**Pull request merged**